projects/m5/003-probability-calculator/js/prob_calculator.js

```diff
@@ -0,0 +1,58 @@
+const cloneDeep = require("lodash.clonedeep");
+
+class Hat {
+  constructor(balls = { black: 1 }) {
+    this.contents = [];
+
+    if (this.#checkInput(balls))
+      throw new Error("The input must be of type object");
```

Comment:
This check is not complete.
It checks that "balls" is an object, but an array is an object too ;-)